tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py

```diff
@@ -0,0 +1,203 @@
+        in_channels = 3
+        out_channels = 2
+        ndim = len(self.dims)
+        ksize = (3,) * ndim
+        self.stride = (2,) * ndim
+        self.pad = (1,) * ndim
+
+        W_scale = numpy.sqrt(1. / functools.reduce(mul, ksize, in_channels))
+        W_shape = (out_channels, in_channels) + ksize
+        self.W = numpy.random.normal(0, W_scale, W_shape).astype(self.W_dtype)
+        self.b = numpy.random.uniform(-1, 1, out_channels).astype(self.x_dtype)
+
+        x_shape = (2, 3) + self.dims
+        self.x = numpy.random.uniform(-1, 1, x_shape).astype(self.x_dtype)
+        gy_shape = (2, 2) + tuple(
+            conv.get_conv_outsize(d, k, s, p, cover_all=self.cover_all)
+            for (d, k, s, p) in zip(self.dims, ksize, self.stride, self.pad))
+        self.gy = numpy.random.uniform(-1, 1, gy_shape).astype(self.x_dtype)
+
+        self.check_forward_options = {}
+        self.check_backward_options = {'dtype': numpy.float64}
+        if self.x_dtype == numpy.float16 or self.W_dtype == numpy.float16:
+            self.check_forward_options = {'atol': 5e-4, 'rtol': 5e-3}
+            self.check_backward_options = {
+                'dtype': numpy.float64, 'atol': 5e-4, 'rtol': 5e-3}
+
+    def check_forward_consistency(self, nobias=False, use_cudnn=False):
+        x_cpu = chainer.Variable(self.x)
+        W_cpu = chainer.Variable(self.W)
+        b_cpu = None if nobias else chainer.Variable(self.b)
+        y_cpu = functions.convolution_nd(
+            x_cpu, W_cpu, b_cpu, stride=self.stride, pad=self.pad,
+            use_cudnn=use_cudnn, cover_all=self.cover_all)
+
+        x_gpu = chainer.Variable(cuda.to_gpu(self.x))
+        W_gpu = chainer.Variable(cuda.to_gpu(self.W))
+        b_gpu = None if nobias else chainer.Variable(cuda.to_gpu(self.b))
+        y_gpu = functions.convolution_nd(
+            x_gpu, W_gpu, b_gpu, stride=self.stride, pad=self.pad,
+            use_cudnn=use_cudnn, cover_all=self.cover_all)
+
+        testing.assert_allclose(
+            y_cpu.data, y_gpu.data.get(), **self.check_forward_options)
+
+    @attr.cudnn
+    def test_forward_consistency(self):
+        self.check_forward_consistency(nobias=False, use_cudnn=True)
+
+    @attr.cudnn
+    def test_forward_consistency_nobias(self):
+        self.check_forward_consistency(nobias=True, use_cudnn=True)
+
+    @attr.gpu
+    def test_forward_consistency_im2col(self):
+        self.check_forward_consistency(nobias=False, use_cudnn=False)
+
+    @attr.gpu
+    def test_forward_consistency_im2col_nobias(self):
+        self.check_forward_consistency(nobias=True, use_cudnn=False)
+
+    def check_backward(self, x_data, W_data, b_data, y_grad, use_cudnn=False):
+        xp = cuda.get_array_module(x_data)
+        if not self.c_contiguous:
+            x_data = xp.asfortranarray(x_data)
+            W_data = xp.asfortranarray(W_data)
+            y_grad = xp.asfortranarray(y_grad)
+            self.assertFalse(x_data.flags.c_contiguous)
```

Comment:
`Flags` class has `f_contiguous` attribute as well (c.f. see [here]( https://github.com/pfnet/chainer/blob/master/cupy/core/flags.pyx#L5)). So it makes the implementation simpler to use this flag.